src/cursor/cur_version.c

```diff
@@ -307,38 +307,37 @@ __curversion_next_int(WT_CURSOR *cursor)
             WT_ERR(__wt_illegal_value(session, page->type));
         }
 
-        /* Get the ondisk value. */
-        ret = __wt_value_return_buf(cbt, cbt->ref, &cbt->upd_value->buf, &cbt->upd_value->tw
-#ifdef HAVE_DIAGNOSTIC
-          ,
-          NULL
-#endif
-        );
-        WT_ERR(ret);
-
-        if (!WT_TIME_WINDOW_HAS_STOP(&cbt->upd_value->tw)) {
-            durable_stop_ts = version_cursor->upd_durable_stop_ts;
-            stop_ts = version_cursor->upd_stop_ts;
-            stop_txn = version_cursor->upd_stop_txnid;
-        } else {
-            durable_stop_ts = cbt->upd_value->tw.durable_stop_ts;
-            stop_ts = cbt->upd_value->tw.stop_ts;
-            stop_txn = cbt->upd_value->tw.stop_txn;
-        }
+        /* Get the ondisk value. We may see an overflow removed cell. Ignore it. */
```

Comment:
Ignoring overflow removed cells doesn't feel like the right choice here. Won't that mean that a record is skipped that shouldn't be?

Having to re-arrange the logic here and in __wt_txn_read to handle WT_RESTART from decoding a cell makes me wonder if the logic should be encapsulated somewhere shared? It's going to be a similar, read from disk, or retry reading the last thing in the update chain.

Reply:
Here if we see the value is overflow removed, we must have seen the same value on the update chain as it is appended to the update chain before the cell is freed.

I don't clearly understand the suggestion. May be we can have a call about it.

Reply:
Another choice here is to still return the removed value to the user. Anyway we still unpack the cell successfully just with a WT_RESTART error returned.